src/expr/subs.h

```diff
@@ -52,7 +52,7 @@ class Subs
   /** Add v -> k for fresh skolem of the same type as v for each v in vs */
   void add(const std::vector<Node>& vs);
   /** Add v -> s to the substitution */
-  void add(const Node& v, const Node& s);
+  virtual void add(const Node& v, const Node& s);
```

Comment:
I don't think we need `virtual` here, it would only make sure that when you do `Subs* subs = new ArithSubs(); subs->add()` that the proper `add()` of the derived class is called.
If we only have `Subs subs; subs.add()` or `ArithSubs subs; subs.add()` we don't need that (and avoid the overhead of dynamic dispatch)

Reply:
You're saying to have the method be overloaded?  This is really confusing in my opinion, perhaps the alternative would be making the derived method `ArithSubs::arithAdd`?